.github/workflows/cli.yml

```diff
@@ -0,0 +1,99 @@
+name: CLI Build
+on:
+  workflow_dispatch:
+  push:
+    branches: [main]
+    paths: 
+      - 'cli/**'
+      - '.github/workflows/cli.yml'
+  pull_request:
```

Comment:
Do we really need to run these jobs on anything except workflow dispatch? I think you could simply this quite a bit by only doing the "release" portions of the jobs.

Reply:
Doing that would mean the flow is never tested except for when we want to make a release. That seems like a recipe for problems.